src/language_features/texlab.rs

```diff
@@ -0,0 +1,96 @@
+use crate::context::Context;
+use crate::position::get_lsp_position;
+use crate::types::{EditorMeta, EditorParams};
+use crate::PositionParams;
+use lsp_types::request::Request;
+use lsp_types::TextDocumentIdentifier;
+use lsp_types::TextDocumentPositionParams;
```

Comment:
please squash such commits, see https://git-rebase.io/ for a guide.
I think other readers don't care about your mistakes ;) and I don't mind force-pushes on ephemeral branches

Reply:
I will sqash all commits once this is satisfactory. This isn't a huge change, so I think a single final commit should suffice.